### PR TITLE
Add etcd-scaling workflows to step-registry

### PR DESCRIPTION
**Pull request by hasbro17**
Adds the `etcd-scaling` workflow for aws/azure/gcp/vsphere which runs the `openshift/etcd/scaling` test suite.

The following presubmits are added to master/4.13/4.12 branches for openshift/cluster-etcd-operator CI:
- e2e-aws-ovn-etcd-scaling
  - mandatory
- e2e-azure-ovn-etcd-scaling
  - optional
- e2e-gcp-ovn-etcd-scaling
  - optional 
- e2e-vsphere-ovn-etcd-scaling
  - optional

Along with nightly jobs for aws/azure/gcp/vsphere for 4.12/4.13

This is required since the etcd vertical scaling test is being moved out of `openshift/conformance/serial` into `openshift/etcd/scaling` in order to reduce disruptions in the serial suite.
See https://github.com/openshift/origin/pull/27444

### CODE REVIEW

**Comment by stbenjam** [comment]
I think it probably makes sense to add a periodic to openshift/release running on some interval - once a day? With the intention of graduating them to payload informers.  You could open it as a separate PR

**Comment by hasbro17** [comment]
Yeah good idea. I was also thinking of periodics to cover all the required platforms. I'll do a follow up once I know these steps are correctly running the new `openshift/etcd/scaling` suite.

**Comment by stbenjam** [comment]
You may also want to move the 4.11 jobs to another PR as well - we'll have to backport the origin PR first before those can work.

File: ci-operator/step-registry/openshift/e2e/aws/etcd-scaling/OWNERS
Comment (tjungblu) [inline ci-operator/step-registry/openshift/e2e/aws/etcd-scaling/OWNERS:8]: also add @wking, I'm sure he also will have to go and plumb some stuff there once in a while.

Comment (tjungblu) [reply ci-operator/step-registry/openshift/e2e/aws/etcd-scaling/OWNERS:8]: feel free to add update in your follow-up PR, let's get this through first ;)

File: ci-operator/step-registry/openshift/e2e/gcp/etcd-scaling/openshift-e2e-gcp-etcd-scaling-workflow.metadata.json
Comment (Elbehery) [inline ci-operator/step-registry/openshift/e2e/gcp/etcd-scaling/openshift-e2e-gcp-etcd-scaling-workflow.metadata.json:14]: Do we need a new line here as the other files ?

**Review by Elbehery — APPROVED** [review]
/lgtm

nice work @hasbro17

**Comment by stbenjam** [comment]
/approve
/hold

Pending merge of the origin side, feel free to remove hold when ready

**Comment by hasbro17** [comment]
Since we're still waiting for CI to go through on the origin PR, I've updated this one to remove the 4.11 presubmits until we get that backported in origin.

@tjungblu if you could retag again that would be great.

**Comment by tjungblu** [comment]
yes sir! 
/lgtm

**Comment by hasbro17** [comment]
@stbenjam I've also lumped in the changes for nightly periodics for the scaling job but let me know if you'd prefer that to be a follow up.
https://github.com/openshift/release/pull/32623/commits/a51a9850508ba3b915b2005f1d830bb10a62dfea

Also since these aren't informing jobs I'm not exactly sure where these would these show up.
Not in the release status or testgrid right?
e.g: https://amd64.ocp.releases.ci.openshift.org/releasestream/4.12.0-0.nightly/release/4.12.0-0.nightly-2022-10-04-081353

Just wondering how we can monitor and get alerted on these jobs, as other repos like the CPSMO would likely break these again in the future.

**Comment by stbenjam** [comment]
I think it's fine to keep it all here.  The periodics on the openshift/release repo  will show up in Sippy, TRT can also configure monitoring for you, maybe on the 7 day pass rate.  Once the job is proven stable, you can graduate to be a release informer.

Drop a card in https://issues.redhat.com/secure/CreateIssue.jspa?pid=12323832&issuetype=17 with 1) names of the periodic jobs, 2) channel to send the alert to, 2) reasonable 7 day threshold (maybe 60% to start?)

**Comment by hasbro17** [comment]
Updated for adding the ovn network as part of the job names.

**Comment by hasbro17** [comment]
/hold

I'm expecting as some early/late tests might fail for the vertical scaling test.
In that case I will likely an additional test suite and workflow that doesn't run those tests so we can use that as a mandatory presubmit on the cluster-etcd-operator repo PRs while we debug those early/late test failures in the periodics.

File: ci-operator/jobs/openshift/cluster-etcd-operator/openshift-cluster-etcd-operator-release-4.12-presubmits.yaml
Comment (hasbro17) [inline ci-operator/jobs/openshift/cluster-etcd-operator/openshift-cluster-etcd-operator-release-4.12-presubmits.yaml:963]: Also not sure why `make jobs` is now marking all jobs as always run. Wasn't doing that the last time I updated these.
Possibly an update in ci-operator but need to confirm what to tweak in the presubmits to not run these. Don't just want to edit these back manually.

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml
Comment (hasbro17) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:1020]: Also not sure why `make jobs` is pruning these now.

**Comment by hasbro17** [comment]
Fixed the extraneous `make job` changes. Now to rehearse the new presubmits again:

/test pj-rehearse

**Comment by stbenjam** [comment]
/lgtm

The kubePodNotReady failures look like something TRT is currently investigating, but I think the rehearsals look good otherwise?

**Comment by hasbro17** [comment]
@stbenjam I think we can merge this now if you want to tag again.

/test pj-rehearse
/unhold

The rehearse jobs look good to me. The test itself is passing on all platforms for both `pull-ci-*` and `periodic-ci-*` jobs, although certain platforms like gcp and vsphere can fail due to the issue of pathological events repeating or some other invariant failures that require further looking into.

Only puzzling thing is the absence of rehearse runs for the 4.13 nightly periodic `gcp` and `azure` jobs i.e:
- rehearse-32623-periodic-ci-openshift-release-master-nightly-4.13-e2e-azure-ovn-etcd-scaling
- rehearse-32623-periodic-ci-openshift-release-master-nightly-4.13-e2e-gcp-ovn-etcd-scaling

https://prow.ci.openshift.org/pr-history/?org=openshift&repo=release&pr=32623

**Pull request merged**
